hivemq-edge/src/frontend/src/modules/Workspace/components/controls/GroupNodesControl.spec.cy.tsx

```diff
@@ -0,0 +1,106 @@
+import { Checkbox, Stack, Text } from '@chakra-ui/react'
+import { MOCK_NODE_ADAPTER, MOCK_NODE_BRIDGE } from '@/__test-utils__/react-flow/nodes.ts'
+import { ReactFlowTesting } from '@/__test-utils__/react-flow/ReactFlowTesting.tsx'
+import useWorkspaceStore from '@/modules/Workspace/hooks/useWorkspaceStore.ts'
+
+import GroupNodesControl from '@/modules/Workspace/components/controls/GroupNodesControl.tsx'
+
+const Wrapper: React.JSXElementConstructor<{ children: React.ReactNode; showDashboard?: boolean }> = ({
+  children,
+  showDashboard = true,
+}) => {
+  const { nodes, onNodesChange } = useWorkspaceStore()
+
+  return (
+    <ReactFlowTesting
+      config={{
+        initialState: {
+          nodes: [
+            { ...MOCK_NODE_ADAPTER, position: { x: 0, y: 0 } },
+            { ...MOCK_NODE_ADAPTER, id: 'idAdapter2', position: { x: 0, y: 0 } },
+            { ...MOCK_NODE_BRIDGE, position: { x: 0, y: 0 } },
+          ],
+        },
+      }}
+      showDashboard={showDashboard}
+      dashboard={
+        <Stack direction="column">
+          {nodes.map((e) => (
+            <Checkbox
+              size="sm"
+              key={e.id}
+              isChecked={e.selected}
+              onChange={() => {
+                onNodesChange([{ id: e.id, type: 'select', selected: !e.selected }])
+              }}
+              data-testid="node-checked"
+            >
+              <Text as="span" data-testid="node-type">
+                {e.type}
+              </Text>{' '}
+              <Text as="span" data-testid="node-group">
+                {e.parentNode || e.parentId ? 'Grouped' : ''}
+              </Text>{' '}
+              <Text as="span" data-testid="node-id">
+                ({e.id})
+              </Text>
+            </Checkbox>
```

Comment:
suggestion(non-blocking): some readability suggestion

```suggestion
          {nodes.map((node) => (
            <Checkbox
              size="sm"
              key={node.id}
              isChecked={node.selected}
              onChange={() => {
                onNodesChange([{ id: node.id, type: 'select', selected: !node.selected }])
              }}
              data-testid="node-checked"
            >
              <Text as="span" data-testid="node-type">
                {node.type}
              </Text>{' '}
              <Text as="span" data-testid="node-group">
                {node.parentNode || node.parentId ? 'Grouped' : ''}
              </Text>{' '}
              <Text as="span" data-testid="node-id">
                ({node.id})
              </Text>
            </Checkbox>
```